crypto/evp/evp_lib.c

```diff
@@ -1045,17 +1065,19 @@ void EVP_MD_CTX_set_update_fn(EVP_MD_CTX *ctx,
 
 void EVP_MD_CTX_set_flags(EVP_MD_CTX *ctx, int flags)
 {
-    ctx->flags |= flags;
+    if (ctx != NULL)
+        ctx->flags |= flags;
 }
 
 void EVP_MD_CTX_clear_flags(EVP_MD_CTX *ctx, int flags)
 {
-    ctx->flags &= ~flags;
+    if (ctx != NULL)
+        ctx->flags &= ~flags;
 }
 
 int EVP_MD_CTX_test_flags(const EVP_MD_CTX *ctx, int flags)
 {
-    return (ctx->flags & flags);
+    return ctx != NULL ? (ctx->flags & flags) : 0;
```

Comment:
Hmmmmm.  Not sure what the failure value should be here.  Zero is a legitimate return.  So is -1.  Is it worthwhile raising a null parameter error on failure?

Reply:
I am not trying to fail here.. just return that it didnt read any flags. (That is the best it can do - You will notice that the set and get return void)

Reply:
I know.  However, flags not set is not distinguishable from error.

Like I wrote: I don't know what a good failure value would be.  Zero isn't terrible but it's not ideal either.

Reply:
I think 0 is OK. If anybody inadvertently calls this with NULL ctx, they'll with high probability call some other EVP_MD_CTX function anyway where the failure can be properly reported.